extra_data/validation.py

```diff
@@ -70,6 +70,21 @@ def check_trainids(self):
                     dataset=ds_path,
                 )
 
+    def _get_index(self, source, group):
+        """returns first and count dataset for specified source.
+        
+        This is slightly different to the same method in FileAccess as it does
+        cut the dataset up to the trainId's dataset length.
+        """
+        ix_group = self.file.file['/INDEX/{}/{}'.format(source, group)]
+        firsts = ix_group['first'][:]
+        if 'count' in ix_group:
+            counts = ix_group['count'][:]
+        else:
+            status = ix_group['status'][:]
+            counts = np.uint64((ix_group['last'][:] - firsts + 1) * status)
+        return firsts, counts
```

Comment:
Copy of `FileAccess._get_index` but returning the full dataset not sliced by trainIds length (required to compare `count` and `first`)